gazebo_plugins/src/gazebo_ros_openni_kinect.cpp

```diff
@@ -306,6 +306,7 @@ bool GazeboRosOpenniKinect::FillPointCloudHelper(
 {
   sensor_msgs::PointCloud2Modifier pcd_modifier(point_cloud_msg);
   pcd_modifier.setPointCloud2FieldsByString(2, "xyz", "rgb");
+  // convert to flat array shape, we need to reconvert later
   pcd_modifier.resize(rows_arg*cols_arg);
```

Comment:
this converts `point_cloud_msg` as height=1 and width=SIZE, so we need reassign the value of height and width for the point cloud msg.